R/create_lkups.R

```diff
@@ -282,15 +284,115 @@ create_lkups <- function(data_dir, versions) {
 #'
 #' @return list
 #' @noRd
-get_vintage_pattern_regex <- function() {
+get_vintage_pattern_regex <- function(vintage_pattern = NULL,
+                                      prod_regex      = NULL,
+                                      int_regex       = NULL,
+                                      test_regex      = NULL
+                                      ) {
+
   list(
-    vintage_pattern = "\\d{8}_\\d{4}_\\d{2}_\\d{2}_(PROD|TEST|INT)$",
-    prod_regex      = "PROD$",
-    int_regex       = "INT$",
-    test_regex      = "TEST$"
+
+    vintage_pattern = ifelse(is.null(vintage_pattern),
+                             "\\d{8}_\\d{4}_\\d{2}_\\d{2}_(PROD|TEST|INT)$",
+                             vintage_pattern),
+
+    prod_regex      = ifelse(is.null(prod_regex),
+                             "PROD$",
+                             prod_regex),
+
+    int_regex       = ifelse(is.null(int_regex),
+                             "INT$",
+                             int_regex),
+
+    test_regex      = ifelse(is.null(test_regex),
+                             "TEST$",
+                             test_regex)
     )
 }
 
+
+#' create vintage call to be parsed into `get_vintage_pattern_regex()`
+#'
+#' @param vintage_pattern either NULL, chracter with regex or list of arguments
+#'   for `get_vintage_pattern_regex()`
+#'
+#' @return list to be parses t `get_vintage_pattern_regex()`
+#' @examples
+#' \dontrun{
+#' vintage_pattern <- NULL
+#' create_vintage_pattern_call(vintage_pattern)
+#'
+#' vintage_pattern <- list("r.*", "", "^hjkhj\\.d")
+#' create_vintage_pattern_call(vintage_pattern)
+#'
+#' vintage_pattern <- c("r.*", "", "^hjkhj\\.d")
+#' create_vintage_pattern_call(vintage_pattern)
+#'
+#' vintage_pattern <- c(vintage_pattern = "r.*", test_regex = "", int_regex =  "^hjkhj\\.d")
+#' create_vintage_pattern_call(vintage_pattern)
+#' }
+create_vintage_pattern_call <- function(vintage_pattern = NULL) {
+
+  #   ______________________________________________________________________
+  #   on.exit                                               ####
+  on.exit({
+
+  })
+
+  #   ____________________________________________________________________________
+  #   Defenses                                                                ####
+  stopifnot( exprs = {
+    class(vintage_pattern) %in% c("NULL",  "list",  "character")
+  }
+  )
+
+  #   _______________________________________________________________________
+  #   Early returns                                             ####
+  if (FALSE) {
+    return()
+  }
+
+  #   ______________________________________________________________________
+  #   Computations                                                      ####
+  vp <-
+    if (is.null(vintage_pattern)) {
+
+      get_vintage_pattern_regex()
+
+    } else {
+
+
+      lf <-
+        formals(get_vintage_pattern_regex) |>
+        names() |>
+        length()
```

Comment:
The R version we use in PROD does not yet support the native pipe I believe. We could update it, but for the time being, let's not use the native pipe